wallet/src/main/scala/org/bitcoins/wallet/models/DLCTable.scala

```diff
@@ -0,0 +1,183 @@
+
+  def remoteFundingInputs: Rep[
+    Vector[(TransactionOutPoint, TransactionOutput)]] =
+    column[Vector[(TransactionOutPoint, TransactionOutput)]](
+      "remoteFundingInputs")
+
+  def feeRate: Rep[SatoshisPerVirtualByte] =
+    column[SatoshisPerVirtualByte]("feeRate")
+
+  def changeSPK: Rep[WitnessScriptPubKey] =
+    column[WitnessScriptPubKey]("changeSPK")
+
+  def remoteChangeSPK: Rep[WitnessScriptPubKey] =
+    column[WitnessScriptPubKey]("remoteChangeSPK")
+
+  def cetSigs: Rep[CETSignatures] = column[CETSignatures]("cetSigs")
+
+  def fundingSigs: Rep[FundingSignatures] =
+    column[FundingSignatures]("fundingSigs")
+
+  def oracleSig: Rep[SchnorrDigitalSignature] =
+    column[SchnorrDigitalSignature]("oracleSig")
+
+  def network: Rep[BitcoinNetwork] = column[BitcoinNetwork]("network")
+
+  private type DLCTuple = (
+      Sha256DigestBE,
+      OracleInfo,
+      ContractInfo,
+      DLCTimeouts,
+      Boolean,
+      ExtPublicKey,
+      CurrencyUnit,
+      CurrencyUnit,
+      Vector[TransactionOutPoint],
+      Vector[(TransactionOutPoint, TransactionOutput)],
+      SatoshisPerVirtualByte,
+      WitnessScriptPubKey,
+      WitnessScriptPubKey,
+      CETSignatures,
+      FundingSignatures,
+      SchnorrDigitalSignature,
+      BitcoinNetwork)
+
+  private val fromTuple: DLCTuple => DLCDb = {
+    case (eventId,
+          oracleInfo,
+          contractInfo,
+          timeouts,
+          isInitiator,
+          remoteExtPubKey,
+          input,
+          remoteInput,
+          fundingInputs,
+          remoteFundingInputs,
+          feeRate,
+          changeSPK,
+          remoteChangeSPK,
+          cetSigs,
+          fundingSigs,
+          oracleSig,
+          network) =>
+      DLCDb(
+        eventId,
+        oracleInfo,
+        contractInfo,
+        timeouts,
+        isInitiator,
+        remoteExtPubKey,
+        input,
+        remoteInput,
+        fundingInputs,
+        remoteFundingInputs,
+        feeRate,
+        changeSPK,
+        remoteChangeSPK,
+        cetSigs,
+        fundingSigs,
+        oracleSig,
+        network
+      )
+  }
+
+  private val toTuple: DLCDb => Option[DLCTuple] = dlc =>
+    Some(
+      (dlc.eventId,
+       dlc.oracleInfo,
+       dlc.contractInfo,
+       dlc.timeouts,
+       dlc.isInitiator,
+       dlc.remoteExtPubKey,
+       dlc.input,
+       dlc.remoteInput,
+       dlc.fundingInputs,
+       dlc.remoteFundingInputs,
+       dlc.feeRate,
+       dlc.changeSPK,
+       dlc.remoteChangeSPK,
+       dlc.cetSigs,
+       dlc.fundingSigs,
+       dlc.oracleSig,
+       dlc.network))
+
+  def * : ProvenShape[DLCDb] =
+    (eventId,
+     oracleInfo,
+     contractInfo,
+     timeouts,
+     isInitiator,
+     remoteExtPubKey,
+     input,
+     remoteInput,
+     fundingInputs,
+     remoteFundingInputs,
+     feeRate,
+     changeSPK,
+     remoteChangeSPK,
+     cetSigs,
+     fundingSigs,
+     oracleSig,
+     network) <> (fromTuple, toTuple)
```

Comment:
Is there a reason `apply` and `unapply` don't work?

Reply:
Don't know just followed the pattern of all the other `XTable.scala` files